File: 백준/Gold/1377. 버블 소트/버블 소트.py
#  **************************************************************************  #
#                                                                              #
#                                                       :::    :::    :::      #
#    Problem Number: 1377                              :+:    :+:      :+:     #
#                                                     +:+    +:+        +:+    #
#    By: hrkim2001 <boj.kr/u/hrkim2001>              +#+    +#+          +#+   #
#                                                   +#+      +#+        +#+    #
#    https://boj.kr/1377                           #+#        #+#      #+#     #
#    Solved: 2025/09/09 23:04:33 by hrkim2001     ###          ###   ##.kr     #
#                                                                              #
#  **************************************************************************  #
import sys
input = sys.stdin.readline

N  = int(input())
A = []
for i in range(N):
    A.append((int(input()), i))

A.sort()
max = 0
for i in range(N):
    temp = A[i][1] - i
    if max < temp:
        max = temp

print(max + 1)

File: 백준/Gold/1377. 버블 소트/버블 소트.java
/* ************************************************************************** */
/*                                                                            */
/*                                                      :::    :::    :::     */
/*   Problem Number: 1377                              :+:    :+:      :+:    */
/*                                                    +:+    +:+        +:+   */
/*   By: hrkim2001 <boj.kr/u/hrkim2001>              +#+    +#+          +#+  */
/*                                                  +#+      +#+        +#+   */
/*   https://boj.kr/1377                           #+#        #+#      #+#    */
/*   Solved: 2026/01/04 22:53:49 by hrkim2001     ###          ###   ##.kr    */
/*                                                                            */
/* ************************************************************************** */
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        
        Node[] A = new Node[N];
        for (int i = 0; i < N; i++) {
            A[i] = new Node(Integer.parseInt(br.readLine()), i);
        }
        Arrays.sort(A);
        
        int max = 0;
        for (int i = 0; i < N; i++) {
            if(A[i].index - i > max) {
                max = A[i].index - i;
            }
        }
        System.out.println(max + 1);
    }

    static class Node implements Comparable<Node> {
        int value;
        int index;

        public Node(int value, int index) {
            this.value = value;
            this.index = index;
        }

        @Override
        public int compareTo(Node o) {
            return this.value - o.value;
        }
    }
}